geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/Strategy.scala

```diff
@@ -127,8 +126,8 @@ trait Strategy {
                     geometryToCover: Geometry,
                     schema: String,
                     featureEncoding: FeatureEncoding,
-                    featureType: SimpleFeatureType) = {
-    if (query.getHints.containsKey(DENSITY_KEY)) {
+                    featureType: SimpleFeatureType) = query match {
+    case densityQuery if densityQuery.getHints.containsKey(DENSITY_KEY) =>
```

Comment:
perhaps use ```case _``` similar to syntax here for consistency on the unused name ```densityQuery```:
https://github.com/locationtech/geomesa/pull/476/files#diff-82e60bfcc5552adf547b65dcc6bea863L241
``` case _: Query if query.getHints.containsKey(DENSITY_KEY) =>```